lib/simp/cli/commands/bootstrap.rb

```diff
@@ -165,6 +168,40 @@ def self.check_for_start_lock
     end
   end
 
+  # Do a quick validation that the code in the malleable SIMP spaces is not
+  # going to cause the compilation to fail out of the box.
+  def self.validate_site_puppet_code
+    info('Validating site puppet code', 'cyan')
+
+    errors = []
+
+    site_pp = File.join(::Utils.puppet_info[:config]['codedir'], 'environments','simp','manifests','site.pp')
+
+    if File.exist?(site_pp)
+      msg = %x{puppet parser validate #{site_pp} 2>&1}
+      unless $?.success?
+        errors << msg.strip
+      end
+    end
+
+    site_module = File.join(::Utils.puppet_info[:config]['codedir'], 'environments','simp','modules','site')
```

Comment:
Why not validate all puppet code?

Reply:
@jeefberkey I actually tried that. It took a LONG time and, in theory, everything else has actually been tested prior to deployment on a standard SIMP system.